File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/node/t2ti_erp_fenix/src/model/pessoa.js
const conexao = require('../config/db_config.js');

const Pessoa = function (pessoa) {
    this.nome = pessoa.nome;
    this.tipo = pessoa.tipo;
    this.site = pessoa.site;
    this.email = pessoa.email;
    this.cliente = pessoa.cliente;
    this.fornecedor = pessoa.fornecedor;
    this.transportadora = pessoa.transportadora;
    this.colaborador = pessoa.colaborador;
    this.contador = pessoa.contador;
};

Pessoa.ConsultarLista = result => {
    conexao.query('SELECT * FROM pessoa', (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        console.log('Lista de objetos', res);
        result(null, res);
    });
};

Pessoa.ConsultarObjeto = (Id, result) => {
    conexao.query('SELECT * FROM pessoa where id = ' + Id, (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        if (res.length) {
            console.log('Objeto localizado', res[0]);
            result(null, res[0]);
            return;
        }

        // objeto não localizado
        result(null, {kind: "not_found"});
    });
};

Pessoa.Inserir = (pessoa, result) => {
    conexao.query('INSERT INTO pessoa SET ?', pessoa, (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        console.log('Objeto inserido', {id: res.insertId, ...pessoa });
        result(null, {id: res.insertId, ...pessoa });
    });
};

Pessoa.Alterar = (Id, pessoa, result) => {
    conexao.query(
        'UPDATE pessoa SET nome = ?, tipo = ?, site = ?, email = ?, cliente = ?, fornecedor = ?, '+
        'transportadora = ?, colaborador = ?, contador = ? where id = ?', 
        [pessoa.nome, pessoa.tipo, pessoa.site, pessoa.email, pessoa.cliente, pessoa.fornecedor,
            pessoa.transportadora, pessoa.colaborador, pessoa.contador, Id], 
        (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        if (res.affectedRows == 0) {
            // objeto não localizado
            result(null, {kind: "not_found"});
            return;
        }

        console.log('Objeto alterado', {id: Id, ...pessoa });
        result(null, {id: Id, ...pessoa });
    });
};

Pessoa.Excluir = (Id, result) => {
    conexao.query('DELETE FROM pessoa where id = ' + Id, (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        if (res.affectedRows == 0) {
            // objeto não localizado
            result(null, {kind: "not_found"});
            return;
        }

        console.log('Objeto excluído com o id = ', Id);
        result(null, res);
    });
};

// Pessoa Jurídica
Pessoa.InserirPessoaJuridica = (pessoaJuridica, result) => {
    conexao.query('INSERT INTO pessoa_juridica SET ?', pessoaJuridica, (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        console.log('Objeto inserido', {id: res.insertId, ...pessoaJuridica });
    });
};


// Contatos
Pessoa.InserirContato = (contato, result) => {
    conexao.query('INSERT INTO pessoa_contato SET ?', contato, (erro, res) => {
        if (erro) {
            console.log('erro:', erro);
            result(null, erro);
            return;
        }

        console.log('Objeto inserido', {id: res.insertId, ...contato });
    });
};

module.exports = Pessoa;

File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/frontend/mvc/lib/src/model/pessoa.dart
class Pessoa implements Comparable<Pessoa> {
  Pessoa();

  String 
        _id,
        _nome,
        _tipo,
        _site,
        _email,
        _cliente,
        _fornecedor,
        _transportadora,
        _colaborador,
        _contador;

  String get id => _id;
  set id(String value){
    if (value == null) value = "";
    _id = value;
  }

  String get nome => _nome;
  set nome(String value){
    if (value == null) value = "";
    _nome = value;
  }

  String get tipo => _tipo;
  set tipo(String value){
    if (value == null) value = "";
    _tipo = value;
  }

  String get site => _site;
  set site(String value){
    if (value == null) value = "";
    _site = value;
  }

  String get email => _email;
  set email(String value){
    if (value == null) value = "";
    _email = value;
  }

  String get cliente => _cliente;
  set cliente(String value){
    if (value == null) value = "";
    _cliente = value;
  }

  String get fornecedor => _fornecedor;
  set fornecedor(String value){
    if (value == null) value = "";
    _fornecedor = value;
  }

  String get transportadora => _transportadora;
  set transportadora(String value){
    if (value == null) value = "";
    _transportadora = value;
  }

  String get colaborador => _colaborador;
  set colaborador(String value){
    if (value == null) value = "";
    _colaborador = value;
  }

  String get contador => _contador;
  set contador(String value){
    if (value == null) value = "";
    _contador = value;
  }

  Pessoa.fromMap(Map m) {
    id = m["id"];
    nome = m["nome"];
    tipo = m["tipo"];
    site = m["site"];
    email = m["email"];
    cliente = m["cliente"];
    fornecedor = m["fornecedor"];
    transportadora = m["transportadora"];
    colaborador = m["colaborador"];
    contador = m["contador"];
  }

  Map<String, dynamic> get toMap {
    return {
      "id": _id,
      "nome": _nome,
      "tipo": _tipo,
      "site": _site,
      "email": _email,
      "cliente": _cliente,
      "fornecedor": _fornecedor,
      "transportadora": _transportadora,
      "colaborador": _colaborador,
      "contador": _contador,
    };
  }
  
  @override
  int compareTo(Pessoa other) => _nome.toLowerCase().compareTo(other._nome.toLowerCase());  
}

